domain-server/src/DomainServerExporter.cpp

```diff
@@ -0,0 +1,456 @@
+{
+    const QString URI_METRICS = "/metrics";
+    const QString EXPORTER_MIME_TYPE = "text/plain";
+
+    qCDebug(domain_server_exporter) << "Request on URL " << url;
+
+    if ( url.path() == URI_METRICS ) {
+        auto nodeList = DependencyManager::get<LimitedNodeList>();
+        QString output = "";
+        QTextStream out_stream(&output);
+
+        nodeList->eachNode([this, &out_stream](const SharedNodePointer& node){
+            generateMetricsForNode(out_stream, node);
+        });
+
+        connection->respond(HTTPConnection::StatusCode200, output.toUtf8(), qPrintable(EXPORTER_MIME_TYPE));
+        return true;
+    }
+
+    return false;
+}
+
+QString DomainServerExporter::escapeName(const QString &name)
+{
+    QRegularExpression invalid_characters("[^A-Za-z0-9_]");
+
+    QString ret = name;
+
+    // If a key is named something like: "6. threads", turn it into just "threads"
+    ret.replace(QRegularExpression("^\\d+\\. "), "");
+    ret.replace(QRegularExpression("^\\d+_"), "");
+
+    // If a key is named something like "z_listeners", turn it into just "listeners"
+    ret.replace(QRegularExpression("^z_"), "");
+
+    // If a key is named something like "lost%", change it to "lost_percent_".
+    // redundant underscores will be removed below.
+    ret.replace(QRegularExpression("%"), "_percent_");
+
+    // change mixedCaseNames to mixed_case_names
+    ret.replace(QRegularExpression("([a-z])([A-Z])"), "\\1_\\2");
+
+    // Replace all invalid characters with a _
+    ret.replace(invalid_characters, "_");
+
+    // Remove any "_" characters at the beginning or end
+    ret.replace(QRegularExpression("^_+"), "");
+    ret.replace(QRegularExpression("_+$"), "");
+
+    // Replace any duplicated _ characters with a single one
+    ret.replace(QRegularExpression("_+"), "_");
+
+    ret = ret.toLower();
+
+    return ret;
+}
+
+void DomainServerExporter::generateMetricsForNode( QTextStream &stream, const SharedNodePointer &node )
+{
+    QString ret = "";
+    QJsonObject statsObject = static_cast<DomainServerNodeData*>(node->getLinkedData())->getStatsJSONObject();
+    QString node_type = NodeType::getNodeTypeName(static_cast<NodeType_t>(node->getType()));
+
+
+    stream << "\n\n\n";
+    stream << "###############################################################\n";
+    stream << "# " << node_type << "\n";
+    stream << "###############################################################\n";
+
+    generateMetricsFromJson(stream, node_type, escapeName(node_type), QHash<QString,QString>(), statsObject);
+
+    QJsonDocument doc(statsObject);
+    ret.append( doc.toJson() );
+
+}
+
+void DomainServerExporter::generateMetricsFromJson(QTextStream &stream, QString original_path, QString path, QHash<QString,QString> labels, const QJsonObject &obj)
+{
+    for(auto iter = obj.constBegin(); iter != obj.constEnd(); ++iter) {
+        auto key = escapeName(iter.key());
+        auto val = iter.value();
```

Comment:
https://github.com/kasenvr/project-athena/blob/master/CODING_STANDARD.md#1212-abbreviations-in-names-should-be-avoided

```suggestion
        auto value = itererator.value();
```

Reply:
I saw this was marked resolved, but the version I see does not contain the fix

Reply:
Because I unwisely clicked the "commit suggestion" button above the previous time, and that broke the build. So I rolled back.

But, I'm going to disagree with this one, this is a bad idea. First, this isn't a public interface, but variables used inside a function. Second, Qt and the STL contain things named 'value' and 'iterator'. It should be abbreviated, precisely to avoid any kind of conflict or confusion.